cmd/plugin/node_server.go

```diff
@@ -31,36 +30,36 @@ const (
 
 type ImagePullStatus int
 
-func NewNodeServer(driver *csicommon.CSIDriver, mounter backend.Mounter, imageSvc cri.ImageServiceClient, secretStore secret.Store, asyncImagePullMount bool) *NodeServer {
-	return &NodeServer{
-		DefaultNodeServer:   csicommon.NewDefaultNodeServer(driver),
-		mounter:             mounter,
-		secretStore:         secretStore,
-		asyncImagePullMount: asyncImagePullMount,
-		mountExecutor: mountexecutor.NewMountExecutor(&mountexecutor.MountExecutorOptions{
-			AsyncMount: asyncImagePullMount,
-			Mounter:    mounter,
-		}),
-		pullExecutor: pullexecutor.NewPullExecutor(&pullexecutor.PullExecutorOptions{
-			AsyncPull:          asyncImagePullMount,
-			ImageServiceClient: imageSvc,
-			SecretStore:        secretStore,
-			Mounter:            mounter,
-		}),
-	}
+func NewNodeServer(driver *csicommon.CSIDriver, mounter backend.Mounter, imageSvc cri.ImageServiceClient, secretStore secret.Store, asyncImagePullTimeout time.Duration) *NodeServer {
+	ns := NodeServer{
+		DefaultNodeServer:     csicommon.NewDefaultNodeServer(driver),
+		mounter:               mounter,
+		imageSvc:              imageSvc,
+		secretStore:           secretStore,
+		asyncImagePullTimeout: asyncImagePullTimeout,
+		asyncImagePuller:      nil,
+	}
+	if asyncImagePullTimeout >= time.Duration(30*time.Second) {
+		klog.Infof("Starting node server in Async mode with %v timeout", asyncImagePullTimeout)
+		ns.asyncImagePuller = remoteimageasync.StartAsyncPuller(context.TODO(), 100, 20)
+	} else {
+		klog.Info("Starting node server in Sync mode")
+		ns.asyncImagePullTimeout = 0 // set to default value
```

Comment:
because the comparison is to 30s, not equality to zero. this ensures that it is either async and >=30s or 0s.

Reply:
this could be avoided if we used a nil check for the async puller or wrapped any checks in a helper method. this avoids the case where 10s would be not be considered async here but may be considered async later due to different criteria being used.

Reply:
for clarity, the `node_server.go` code currently DOES use nil check to see if running in async mode but someone could also have read the Cobra parameter description and tried to use the async timeout DURATION itself. this could be error prone over time. instead, i'm forcing consistency here.